flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -409,7 +416,7 @@ private ListShardsResult listShards(String streamName, @Nullable String startSha
 		int attemptCount = 0;
 		// List Shards returns just the first 1000 shard entries. Make sure that all entries
 		// are taken up.
-		while (listShardsResults == null) { // retry until we get a result
+		while (attemptCount <= listShardsMaxAttempts && listShardsResults == null) { // retry until we get a result
```

Comment:
The earlier contract was to wait till we get a result. https://issues.apache.org/jira/browse/FLINK-10020 does not talk about breaking this contract. I personally believe maxAttemptCount is better since listShard works in a periodic thread and we are bound to try again after 'X' seconds. Just wanted to point this out. I like this approach better.

Reply:
I too think that this is better since it provides the user more flexibility. Setting the retry count to max practically achieves the previous behavior. Perhaps we should up the default retry count?